Flora/Flora/View Controllers/UserProfileViewController.m

```diff
@@ -56,29 +65,18 @@ - (__kindof UICollectionViewCell *)collectionView:(UICollectionView *)collection
     ProfileBoardCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"ProfileBoardCell" forIndexPath:indexPath];
     cell.delegate = self;
     
-    PFQuery *query = [PFQuery queryWithClassName:@"Board"];
-    [query whereKey:@"name" equalTo:self.user[@"boards"][indexPath.row]];
-    [query whereKey:@"user" equalTo:self.user.username];
-    query.limit = 1;
-        
-    [query findObjectsInBackgroundWithBlock:^(NSArray * _Nullable results, NSError * _Nullable error) {
-        if(error) {
-            NSLog(@"Error getting board: %@", error.localizedDescription);
-        } else  if (results.count > 0) {
-            Board *board = (Board *)results[0];
-            cell.board = board;
-            cell.boardName.text = board.name;
-            cell.numPlants.text = [[NSString stringWithFormat:@"%li",  board.plantsArray.count] stringByAppendingString:@" plants"];
-            
-            [self setBoardCoverImage:board.plantsArray[0] forCell:cell];
-        }
-    }];
+    Board *board = self.boardsArray[indexPath.row];
+    cell.board = board;
+    cell.boardName.text = board.name;
+    cell.numPlants.text = [[NSString stringWithFormat:@"%li",  board.plantsArray.count] stringByAppendingString:@" plants"];
```

Comment:
These two setters are using data from `board`.

A better pattern would be to define a custom setter for `board` in `ProfileBoardCell` and do this logic internally.

Then the external API is just
`cell.board = board;`

Here you are exposing view logic to the view controller, rather than centralizing it in the view.